worker/backend/spec/spec.go

```diff
@@ -109,19 +99,61 @@ func OciSpecBindMounts(bindMounts []garden.BindMount) (mounts []specs.Mount, err
 	return
 }
 
+// OciIDMappings provides the uid/gid mappings for user namespaces (if
+// necessary, based on `privileged`).
+//
+func OciIDMappings(privileged bool, max uint32) []specs.LinuxIDMapping {
+	if privileged {
+		return []specs.LinuxIDMapping{}
+	}
+
+	return []specs.LinuxIDMapping{
+		{ // "root" inside, but non-root outside
+			ContainerID: 0,
+			HostID:      max,
+			Size:        1,
+		},
+		{ // anything else, not root inside & outside
+			ContainerID: 1,
+			HostID:      1,
+			Size:        max - 1,
+		},
```

Comment:
I'm not totally getting this part - why do we map to IDs `1 -> (max-1)` on the Host? I get that they're all just non-root uids, but does this not conflict with other containers' mappings (or are they all isolated)?

Reply:
yeah, the mappings *will* indeed conflict: from the host perspective, it's
totally expected (in this implementation) that `user 123` in `containerA` (which
maps to `user 123` in the `host` user namespace) will be the same as `user 123`
in `containerB`.

While *not* overlapping is something that we *could* do,  that's the current
behavior of [`gdn`] - thus, wouldn't really fall into this PR, if we wanted to
get to it.

Also, [`baggageclaim`] performs the same assumptions (see
[`uidgid/mapper_linux.go`], so we'd probably need to make some changes there
too.

[`gdn`]: https://github.com/cloudfoundry/guardian
[`baggageclaim`]: https://github.com/concourse/baggageclaim
[`uidgid/mapper_linux.go`]: https://github.com/concourse/baggageclaim/blob/df945db3059d4f9ace9284994f37b625c3fd864e/uidgid/mapper_linux.go#L13-L41